### PR TITLE
3 new factions

### PR DESCRIPTION
**Pull request by zonkmachine**
https://github.com/pioneerspacesim/pioneer/pull/5252 in one commit rebased on master.

3 factions removed:
Blood Worlds
Continuity Horde
Continuity Empire

3 replacement factions:
Druello Pirates
Tolan kingdom
Julian science

Closes https://github.com/pioneerspacesim/pioneer/pull/5252

### CODE REVIEW

**Review by impaktor — COMMENTED** [review]
```
<impaktor> If the factions are hand written, I'd expect the numbering to
           not be "077...", but 004 or similar
```
I assume changing pre-fix number changes appearance in system view?

**Comment by zonkmachine** [comment]
> ```
> <impaktor> If the factions are hand written, I'd expect the numbering to
>            not be "077...", but 004 or similar
> ```
> 
> I assume changing pre-fix number changes appearance in system view?

(_I'm not sure I fully understand this, but..._) I don't think so. It's just the name of the file. I see no reason to change this.

**Comment by Web-eWorks** [comment]
Files are loaded by disk iteration order, yes. There are potential knock-on effects on system generation consistency, but I haven't fully audited the code to determine whether that is the case or not. Proceed with a rename... cautiously.

File: data/factions/079_Julian_science.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
+-- Copyright © 2008-2021 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local f = Faction:new('Julian Science Commune')
```
[the remark this answers is not in the document]
Comment (Web-eWorks) [reply data/factions/079_Julian_science.lua:4]: I agree, but I cannot come up with one off the top of my head.

**Comment by Web-eWorks** [comment]
Actually, given that we've already merged a savebump commit, proceed with a rename with great abandon! :smile:

**Comment by zonkmachine** [comment]
The only think I can think of remaining is the question of a new name for Julian Science. I'm personally fine with the PR in it's current state but more than one person opposed to that name so we should look for a new one I guess. Any suggestions? Anyone?

**Comment by zonkmachine** [comment]
I see that I've been inconsistent with capitalization in the file/faction names. Which is preferred.
Druello pirates, Tolan kingdom, Julian science.
or
Druello Pirates, Tolan Kingdom, Julian Science?

**Comment by Web-eWorks** [comment]
All faction names are Proper Nouns, and should be capitalized as such. Everything except linking verbs (e.x: Polity of Location, Place-by-the-Sea) should be capitalized.

File: data/factions/079_Julian_science.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (Web-eWorks) [reply data/factions/079_Julian_science.lua]: The system that this faction has as its homeworld is currently named Tolisay... and is entirely uninhabited.

I've found a bug with how certain data members are filled when generating binary systems that I need to fix before I could make a custom system for the faction, but I propose that the faction be renamed the "Tolisian Scientific Community" or "Tolisian Scientific Society" and a custom system accompany the new faction.

Similarly, the other factions should be checked to ensure their assigned homeworlds are congruent with the faction data, with custom systems contributed as necessary.

File: data/factions/077_Tolan_Kingdom.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
+-- Copyright © 2008-2024 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local f = Faction:new('Tolan Kingdom')
+	:description_short('Tolan Kingdom')
+	:description('The Tolan Kingdom, famous (or infamous, depending on who you ask) for repelling Haber invasion forces 500 years ago.')
+	:homeworld(2,-9,-19,2,4) -- edge of Habers sphere of influence
+	:foundingDate(2337) -- as per timeline
+	:expansionRate(0) -- a single system only, but quite populous
```
Comment (Web-eWorks) [inline data/factions/077_Tolan_Kingdom.lua:9]: Per current lore / IRC discussion (CC @bszlrd) the Tolan Kingdom is a small, insular faction spanning a few systems. Either the expansionRate variable should be a small non-zero value, or a few specific systems assigned to the faction (probably surrounding Tolan but not intruding too far into Haber space).

Comment (zonkmachine) [reply data/factions/077_Tolan_Kingdom.lua]: OK. 0.01 gives the Tolans a dozen or so system.

**Comment by zonkmachine** [comment]
I bumped the Tolans expansion rate to 0.01. That gives them a dozen systems. I had a peek in the sector/system viewers and it's home world isn't inhabited. It has some three gas giants with no moons in it an nothing else.

File: data/factions/079_Julian_science.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (zonkmachine) [reply data/factions/079_Julian_science.lua]: OK. I bumped into the uninhabited homeworlds too. Julians are now Tolisians.

**Comment by zonkmachine** [comment]
> So it is ripe for a custom system? (not necesseraly in this PR)

Yes. And I not in this PR.

**Review by Web-eWorks — APPROVED** [review]
I think this PR is in a good state to merge, once the git history is cleaned up and fixup commits are squashed. I'll try to work on the system editor / generation issues today so custom systems can be made for each of these factions.

**Comment by impaktor** [comment]
I have a branch with half done Pirate system, and Pirate Fuel Club: "Gas Guzzler", I don't quite remember if I added something more. My thinking is a pirate system is sort of hidden. What would an entire pirate faction be like? Sounds more like anarchy to me. But then again, there's not that much parameters for factions to tweak to make them distinct. 
End of random thought.

**Comment by zonkmachine** [comment]
> [Here's the note on Tolan](https://github.com/bszlrd/Pioneer-Worldbuilding/blob/master/Notes/Tolan%20history%20and%20society.md).

Nothing in that link.

> Pirate Fuel Club: "Gas Guzzler",

:skull_and_crossbones:

**Comment by Web-eWorks** [comment]
>  My thinking is a pirate system is sort of hidden. What would an entire pirate faction be like? Sounds more like anarchy to me.

I agree, it definitely seems weird in the abstract that a pirate organization would become an interstellar faction - however, out in the middle of independent space with little to no government larger than a planetary administration or maybe a system-wide kingdom I could see various enterprising pirates "going legit" and setting up some type of protection racket or feudal-analogue power structure with various worlds being nominally "ruled" by a nomadic interstellar navy.

**Comment by Web-eWorks** [comment]
Amusingly, the system at `4, -1, 9` is actually called Torisay, not Tolisay. That's on me, but I think I like Tolisay better.

Either the faction will have to be renamed (again), or I'll have to find some way to implement custom systems overriding an existing random system while also changing the name (which is used to detect if the system is overridden)...

**Pull request merged**